### PR TITLE
GET api route for all questions and questions by id

### PR DESCRIPTION
**Pull request by cph-kiwi**
# Description

We want to be able to get questions with specific criteria so that we can show those questions. I created a GET request for all the questions and to retrieve a question by id. I'd like to add more specific criteria about searching for questions created before or after a certain date, and for searching for questions with specific words in. However, the issue instructed to break the PR's up a bit so that they aren't huge. I can add more once this has been approved.

Fixes # https://github.com/HackYourFuture-CPH/Guide-IT/issues/29

# How to test?

Once you have run the migration and seeding to your local database, open this branch and run npm run dev. Go to localhost:3000/api/documentation. You will see two GET requests for questions. You can run the tests from there, to retrieve all the questions and a question by its id.

# Checklist

- [x ] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Comment by cph-kiwi** [comment]
The typescript camelcase plugin does not seem to be supported in the current version of eslint. So I removed the problematic line from the config.

File: src/server/api/routes/questions.router.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/:id', (req, res, next) => {
+  questionsController
+    .getQuestionById(req.params.id)
+    .then((result) => res.json(result))
```
Comment (qamarfj) [inline src/server/api/routes/questions.router.js:58]: I think user expecting single object but this code sending an array having single object as an element.

**Comment by Poorooshka** [comment]
Hi Beth, it's best to keep the PR name as simple as possible. So I this it's better to remove the word "created"

File: src/server/api/routes/questions.router.js (hunk 1 above)
Comment (cph-kiwi) [reply src/server/api/routes/questions.router.js:58]: I followed the modules example, as I believe this is what is expected as output. Anyone else want to offer an opinion? I'm happy to change it, but I'd rather wait until I hear some other opinions.

**Review by qamarfj — APPROVED** [review]

File: src/server/api/routes/questions.router.js (hunk 1 above)
Comment (TITHI1244) [reply src/server/api/routes/questions.router.js:58]: I agree with Beth. I also think an array of questions make more sense to me :)

**Review by TITHI1244 — APPROVED** [review]
Nice and clean code. Good job @cph-kiwi :)

File: src/server/api/controllers/questions.controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  try {
+    const questions = await knex('questions')
+      .select('questions.id as id', 'question')
+      .where({ id });
+    if (questions.length === 0) {
+      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    }
+    return questions;
```
Comment (bytewiz) [inline src/server/api/controllers/questions.controller.js:22]: why is it named `questions` and not `question`? it is supposed to only ever return one question right? :)

File: src/server/api/routes/questions.router.js (hunk 1 above)
Comment (bytewiz) [reply src/server/api/routes/questions.router.js:58]: Here I would expect a single object (question) returned not an array as it is one question we get

**Comment by cph-kiwi** [comment]
@bytewiz I made changes as a result of your suggestions. I pulled from develop branch before pushing, and then had to bug fix before I could commit changes, so a couple of extra files there now. But it's ready to approve.

**Pull request merged**